performance-tracking-core/src/main/java/com/rakuten/tech/mobile/perf/core/MeasurementBuffer.java

```diff
@@ -18,40 +19,51 @@
 
   Measurement next() {
     int id = nextTrackingId.getAndIncrement();
+    int nextIndex = indexForId(id);
 
-    if (id == 0) {
-      // Ensure id is not zero as zero has special purpose
-      id = nextTrackingId.getAndIncrement();
-    }
-
-    int index = id % SIZE;
-    if (index < 0) {
-      index += SIZE;
-    }
-
-    Measurement m = at[index];
-
-    if (m.trackingId != 0) {
+    if (hasMeasurement(nextIndex)) {
       nextTrackingId.getAndDecrement();
       return null;
     }
 
-    m.trackingId = id;
+    Measurement measurement = at[nextIndex];
+    measurement.trackingId = id;
 
-    return m;
+    return measurement;
   }
 
   Measurement getByTrackingId(int trackingId) {
-    int index = trackingId % SIZE;
-    if (index < 0) {
-      index += SIZE;
+    Measurement measurement = at[indexForId(trackingId)];
+    if (measurement.trackingId == trackingId) {
+      return measurement;
     }
 
-    Measurement m = at[index];
-    if (m.trackingId == trackingId) {
-      return m;
+    return null;
+  }
+
+  int count(int startId) {
+    int id = nextTrackingId.get();
+    int endIndex = indexForId(id);
+
+    int count;
+
+    if (startId == endIndex && hasMeasurement(endIndex)) {
```

Comment:
The main reason I changed this is because, if the buffer isn't full, then `buffer.next()` creates an empty measurement which never gets used. This means that calling `count` would actually change the buffer (and the newly created measurement isn't included in the count). 

I think it will be better if I change `hasMeasurement` to instead check `isFull`.